src/address.ts

```diff
@@ -259,3 +268,70 @@ export class Address {
         return this.toHex().startsWith(SMART_CONTRACT_HEX_PUBKEY_PREFIX);
     }
 }
+
+export class AddressComputer {
+    private readonly numberOfShardsWithoutMeta: number;
+
+    constructor(numberOfShardsWithoutMeta?: number) {
+        this.numberOfShardsWithoutMeta = numberOfShardsWithoutMeta || CURRENT_NUMBER_OF_SHARDS_WITHOUT_META;
+    }
+
+    computeContractAddress(deployer: IAddress, deploymentNonce: bigint): Address {
+        let initialPadding = Buffer.alloc(8, 0);
```

Comment:
`const` where possible. Here, and below.

Reply:
Done!